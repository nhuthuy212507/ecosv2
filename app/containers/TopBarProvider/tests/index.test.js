/**
 * Test the HomePage
 */

import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import { browserHistory, BrowserRouter as Router } from 'react-router-dom';

import {
  changeDesktopSidebarVisibility,
  changeMobileSidebarVisibility,
} from 'containers/SideBarProvider/actions';
import { TopBarProvider, mapDispatchToProps } from '../index';
import configureStore from '../../../configureStore';

describe('<TopBarProvider />', () => {
  let store;

  beforeAll(() => {
    store = configureStore({}, browserHistory);
  });

  it('should render and match the snapshot', () => {
    const changeDesktopSidebarVisibilitySpy = jest.fn();
    const changeMobileSidebarVisibilitySpy = jest.fn();
    const wrapper = shallow(
      <Provider store={store}>
        <Router>
          <TopBarProvider
            onChangeDesktopSidebarVisibility={changeDesktopSidebarVisibilitySpy}
            onChangeMobileSidebarVisibility={changeMobileSidebarVisibilitySpy}
          />
        </Router>
      </Provider>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  describe('mapDispatchToProps', () => {
    describe('onChangeDesktopSidebarVisibility', () => {
      it('should be injected', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        expect(result.onChangeDesktopSidebarVisibility).toBeDefined();
      });

      it('should dispatch onChangeDesktopSidebarVisibility when called', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        result.onChangeDesktopSidebarVisibility();
        expect(dispatch).toHaveBeenCalledWith(changeDesktopSidebarVisibility());
      });
    });

    describe('onChangeMobileSidebarVisibility', () => {
      it('should be injected', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        expect(result.onChangeMobileSidebarVisibility).toBeDefined();
      });

      it('should dispatch onChangeMobileSidebarVisibility when called', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        result.onChangeMobileSidebarVisibility();
        expect(dispatch).toHaveBeenCalledWith(changeMobileSidebarVisibility());
      });
    });
  });
});
