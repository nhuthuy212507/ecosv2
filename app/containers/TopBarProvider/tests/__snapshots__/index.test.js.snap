// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TopBarProvider /> should render and match the snapshot 1`] = `ShallowWrapper {}`;
